imports/wasi_snapshot_preview1/wasmedge.go

```diff
@@ -326,16 +326,32 @@ func (m *Module) WasmEdgeSockAddrInfo(ctx context.Context, name String, service
 }
 
 func (m *Module) wasmEdgeGetSocketAddress(b wasmEdgeAddress, port int) (sa wasi.SocketAddress, ok bool) {
+	// V2 addresses.
 	if len(b) == 128 {
 		switch wasi.ProtocolFamily(binary.LittleEndian.Uint16(b)) {
 		case wasi.InetFamily:
-			b = b[2:6]
+			b = b[2:6] // fallthrough to v1 parser below
 		case wasi.Inet6Family:
-			b = b[2:18]
+			b = b[2:18] // fallthrough to v1 parser below
+		case wasi.UnixFamily:
+			b = b[2:]
+			n := 0
+			for ; n < len(b); n++ {
+				if b[n] == 0 {
+					break
+				}
+			}
```

Comment:
Maybe a shorter expression could be:
```suggestion
			for n < len(b) && b[n] != 0 {
				n++
			}
```